src/main/kotlin/calculator/domain/Calculator.kt

```diff
@@ -0,0 +1,25 @@
+package calculator.domain
+
+class Calculator(private val expression: Expression) {
+
+    fun calculate(): Int {
+        if (expression.value.isEmpty()) {
+            return 0
+        }
+
+        val operands = expression.value.mapNotNull { it as? ExpressionElement.OperandElement }
+        val operators = expression.value.mapNotNull { it as? ExpressionElement.OperatorElement }
```

Comment:
map 보다는 `filterInstance<T>()` 함수가 더 의도하신 방향에 맞는 것 같아요. 😃
```suggestion
        val operands = expression.value.filterIsInstance<ExpressionElement.OperandElement>()
        val operators = expression.value.filterIsInstance<ExpressionElement.OperatorElement>()
```